morphia/src/main/java/org/mongodb/morphia/AdvancedDatastore.java

```diff
@@ -247,6 +254,16 @@
     void setDecoderFact(DBDecoderFactory fact);
 
     /**
+     * Returns the {@see MongoCollection} for the given name
+     *
+     * @param <T>   the type of the MongoCollection
+     * @param name  the name of the collection
+     * @param clazz the type of the collection
+     * @return the collection
+     */
+    <T> MongoCollection<T> getMongoCollection(String name, Class<T> clazz);
```

Comment:
Let's leave it for when we execute that plan.  For now keep it out of the public API.